ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Ledger/Tables.hs

```diff
@@ -0,0 +1,388 @@
+--        , utxoSet = Map.fromList [(\'a\', 100), (\'b\', 100), ...]
+--        }
+--    , theTables = 'EmptyMK'
+--    })
+-- @
+--
+-- Using these functions we can extract the data from the Ledger /ledger state/
+-- for us Consensus to manipulate, and we can then inject it back so that we
+-- provide the expected data to the ledger. Note that the Ledger rules for
+-- applying a block are defined in a way that it only needs the subset of the
+-- UTxO set that the block being applied will consume. See [the @DbChangelog@
+-- documentation for block
+-- application](Ouroboros-Consensus-Storage-LedgerDB-DbChangelog.html#g:applying).
+--
+-- Now using 'Ouroboros.Consensus.Ledger.Tables.Utils.calculateDifference', we
+-- can compare two (successive) t'Ouroboros.Consensus.Ledger.Basics.LedgerState's
+-- to produce differences:
+--
+-- @
+-- 'Ouroboros.Consensus.Ledger.Tables.Utils.calculateDifference'
+--   (LedgerState {
+--       ...
+--     , theTables = 'ValuesMK' (Map.fromList [(\'a\', 100), (\'b\', 100)])
+--     })
+--   (LedgerState {
+--       ...
+--     , theTables = 'ValuesMK' (Map.fromList [(\'a\', 100), (\'c\', 200)])
+--     })
+-- ==
+--  'TrackingMK'
+--    (Map.fromList [(\'a\', 100),      (\'c\', 200)])
+--    (Map.fromList [(\'b\', [Delete]), (\'c\', [Insert 200])])
+-- @
+--
+-- This operation provided a 'TrackingMK' which is in fact just a 'ValuesMK' and
+-- 'DiffMK' put together.
+--
+-- We can then use those differences to /forward/ a set of values, so for
+-- example (taking the example above):
+--
+-- @
+-- let state1 = LedgerState {
+--                 ...
+--               , theTables = 'ValuesMK' (Map.fromList [(\'a\', 100), (\'b\', 100)])
+--              }
+--     state2 = LedgerState {
+--                ...
+--              , theTables = 'ValuesMK' (Map.fromList [(\'a\', 100), (\'c\', 200)])
+--              }
+--     state3 = LedgerState {
+--                ...
+--              , theTables = 'ValuesMK' (Map.fromList [])
+--              }
+-- in
+--   'Ouroboros.Consensus.Ledger.Tables.Utils.applyDiffs' state3 ('Ouroboros.Consensus.Ledger.Tables.Utils.forgetTrackingValues' $ 'Ouroboros.Consensus.Ledger.Tables.Utils.calculateDifference' state1 state2)
+-- ==
+--  LedgerState {
+--      ...
+--    , theTables = 'ValuesMK' (Map.fromList [(\'c\', 200)])
+--    }
+-- @
+--
+-- Notice that we produced differences for @\'b\'@ and @\'c\'@, but as the input
+-- state (@state3@) didn't contain @\'b\'@ the only difference that was applied
+-- was the one of @\'c\'@.
+--
+-- Also when applying a block that contains some transactions, we can produce
+-- 'LedgerTable's of @KeysMK@, by gathering the txins required by the
+-- transactions:
+--
+-- @
+-- 'Ouroboros.Consensus.Ledger.Abstract.getBlockKeySets' (Block {..., txs = [Tx { input = [\'a\', \'b\'], outputs = [\'c\', \'d\'] }]})
+--  == 'KeysMK' (Set.fromList [\'a\', \'b\'])
+-- @
+--
+-- We shall use those later on to read the txouts from some storage (which will
```

Comment:
Agreed, using ledger tables instead of ledger state in the example might simplify the examples that don't involve `stowLedgerTables`/`unstowLedgerTables`